core/data/repo.go

```diff
@@ -78,6 +89,17 @@ func (m *CFNTrackerRepository) CreateSession(ctx context.Context, userId string)
 	return &sesh, nil
 }
 
+func (m *CFNTrackerRepository) GetSession(ctx context.Context, userId string) (*Session, error) {
```

Comment:
Could rename this to GetLastSession. The current name suggests that the user gets to decide on which session.